compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSClassifierReferenceLiteImplForJava.kt

```diff
@@ -0,0 +1,39 @@
+package com.google.devtools.ksp.symbol.impl.java
+
+import com.google.devtools.ksp.ExceptionMessage
+import com.google.devtools.ksp.symbol.KSClassifierReference
+import com.google.devtools.ksp.symbol.KSTypeArgument
+import com.google.devtools.ksp.symbol.Location
+import com.google.devtools.ksp.symbol.NonExistLocation
+import com.google.devtools.ksp.symbol.Origin
+import com.google.devtools.ksp.symbol.impl.KSObjectCache
+import com.intellij.psi.PsiAnnotation
+import com.intellij.psi.PsiMethod
+
+class KSClassifierReferenceLiteImplForJava(override val parent: KSTypeReferenceLiteJavaImpl) : KSClassifierReference {
+    companion object : KSObjectCache<KSTypeReferenceLiteJavaImpl, KSClassifierReference>() {
+        fun getCached(parent: KSTypeReferenceLiteJavaImpl) = KSClassifierReferenceLiteImplForJava.cache
+            .getOrPut(parent) { KSClassifierReferenceLiteImplForJava(parent) }
+    }
+    override val qualifier: KSClassifierReference? = null
```

Comment:
for normal classifier implementations,`qualifier` is non null only if there is a qualifier defined in the code for that classifier reference, e.g. `val a: kotlin.collections.List`, therefore I don't want to go too far in qualifiers here to respect source code as well as align with behavior from other implementations.

Reply:
actually this might not be trivial, the key has be to include `parent` instead of just `name`, because `ClassifierReference` is associated with a symbol in source code.